### DISCUSSION TITLE
Color of GIF Image generated using ImageMagick 6.9 differ from GIF image generated using ImageMagick 6.2

### DISCUSSION THREAD

**Discussion by VedanthBabu** [ImageMagick/ImageMagick#7812] [Help]
### ImageMagick version

6.9.12-95-portable-Q16-HDRI-x64

### Operating system, version and so on

Windows

### Description

I am trying to upgrade ImageMagick version from 6.2 to 6.9. I am using ImageMagick to generate GIF from EPS or TIFF. After installing 6.9 version of ImageMagick I see generated images have different color to images generated in version 6.2. I am using following command.
convert test.eps -profile "USWebCoatedSWOP.icc" -profile "sRGB.icm" test.gif

Below is an example of color difference.

color in 6.2
![image](https://github.com/user-attachments/assets/deddacbc-5969-4e14-b968-6c2cc1030b34)

color in 6.9

![image](https://github.com/user-attachments/assets/118a8391-10cb-41cb-b81a-8f030431dd14)

How can I get colors similar to images generated using version 6.2 using ImageMagick version 6.9.

Any information/advice for this problem would be a great news.
Thank you for your consideration!

**Comment by snibgo** [comment]
In your command...
```
convert test.eps -profile "USWebCoatedSWOP.icc" -profile "sRGB.icm" test.gif
```
... if test.eps has no embedded ICC profile, then it will be *assigned* the USWebCoatedSWOP. This means that the pixels are assumed to be encoded as USWebCoatedSWOP, and that assumption may be wrong.

V6.2 is very very very old. The colour difference may be due to bug(s) or limitation(s) in the very old code, either in ImageMagick or an underlying library such as LCMS.

I doubt that IM options can be used to simulate the old behaviour.

You might test the round-trip from sRGB to USWebCoatedSWOP and back to sRGB, like this, where "toes.png" is an ordinary sRGB photo with no embedded ICC profile.
```
magick toes.jpg -strip -profile sRGB.icc -profile USWebCoatedSWOP.icc -profile sRGB.icc x.png

magick compare -metric RMSE toes.jpg x.png NULL:
349.773 (0.0053372)
```
The RMSE is 0.53372%, which is not a visible change, so the round-trip gets us back where we started.

**Comment by fmw42** [comment]
Older versions of IM had RGB and sRGB swapped.

You might also try extracting the image from the eps file using pdfimages or some similar tool for windows.
